### ISSUE TITLE
getty in init errors trying to modify securetty

### ISSUE THREAD

**Issue by justincormack** [linuxkit/linuxkit#2097]
If you use `getty` in the `init` section I just happened to notice it says:

```
/usr/bin/rungetty.sh: line 1: can't create /etc/securetty: Read-only file system
```

Thats probably not a terrible disaster, but maybe we should try to fix it...

**Comment by justincormack** [comment]
In some ways I think if you can modify securetty its kind of pointless... Maybe we should have a fixed list, and you change it in `files` if it doesnt work for you?

**Comment by justincormack** [comment]
cc @deitch

**Comment by ijc** [comment]
I noticed this yesterday and half wondered if it should be inverted, i.e. only start getty's on things listed in `securetty` by default.

**Comment by deitch** [comment]
> can't create /etc/securetty: Read-only file system

when you run `linuxkit/init` first, doesn't it mount `/` (in which `/etc` resides) read-only?

> only start getty's on things listed in securetty by default.

Originally, we just copied what `linuxkit/init` did, but more than happy to have a better solution.
The challenge is that we have been assuming that wherever there is a `console=...` line in `/proc/cmdline` is, by definition, where you want getty to operate.

Looks like you both are saying to have a default `/etc/securetty` in `linuxkit/getty`, and modify it with `files:` if someone wants to override it. I am OK with that, but how would it work in practice?

* As `service`: look for `/etc/securetty` as created by `files:` from the host (already mounted `/etc:/hostroot/etc` for cases like this); if it exists, override, if not, use the default in the `getty` image
* As `init`: "host" `/etc/` won't exist in the usual place, but `files:`-created `/etc/securetty` would override it, so it would just inherit it.

Is that the idea? If we three are in agreement on it, I am happy to change `linuxkit/getty`.

**Comment by deitch** [comment]
is there agreement on this one? Where do we stand? Do we think that reading `/etc/securetty` is better than `/proc/cmdline` for `console=` entries? I am a little worried about people who might have modified it, but then again, we have a default for both `/etc/securetty` and the `cmdline` console entries, so we match them.

**Comment by justincormack** [comment]
I don't think we should attempt to modify `securetty`. If possible print out a warning that it is going to fail. Overall not sure it makes a lot of sense as a security mechanism for us.

**Comment by deitch** [comment]
> I don't think we should attempt to modify securetty. If possible print out a warning that it is going to fail. Overall not sure it makes a lot of sense as a security mechanism for us.

So... what are you suggesting we do? Leave as is? Follow `securetty` and don't change it? Follow console?

**Comment by justincormack** [comment]
Follow the console, leave `securetty` as is, but print an error if the console why are trying to use is not is `securetty` so user gets some idea why they are not getting a console

**Comment by deitch** [comment]
I like that. Do we need to send it explicitly to the console via `2>/dev/console`? `services` do not automatically send their stdout/stderr to console?

**Comment by deitch** [comment]
Closed via #2179
